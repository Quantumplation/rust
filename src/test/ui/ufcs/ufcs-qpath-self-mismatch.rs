use std::ops::Add;

fn main() {
    <i32 as Add<u32>>::add(1, 2);
    //~^ ERROR cannot add `u32` to `i32`
    <i32 as Add<i32>>::add(1u32, 2);
    //~^ ERROR arguments to this function are incorrect
    <i32 as Add<i32>>::add(1, 2u32);
    //~^ ERROR arguments to this function are incorrect
}
