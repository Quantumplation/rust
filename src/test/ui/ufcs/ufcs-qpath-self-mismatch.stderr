error[E0277]: cannot add `u32` to `i32`
  --> $DIR/ufcs-qpath-self-mismatch.rs:4:5
   |
LL |     <i32 as Add<u32>>::add(1, 2);
   |     ^^^^^^^^^^^^^^^^^^^^^^ no implementation for `i32 + u32`
   |
   = help: the trait `Add<u32>` is not implemented for `i32`
   = note: required by `add`

error[E0308]: arguments to this function are incorrect
  --> $DIR/ufcs-qpath-self-mismatch.rs:6:5
   |
LL |     <i32 as Add<i32>>::add(1u32, 2);
   |     ^^^^^^^^^^^^^^^^^^^^^^^----^^^^
   |                            |
   |                            expected `i32`, found `u32`
   |
help: provide an argument of the correct type
   |
LL |     ({i32}, 2);
   |     ^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/ufcs-qpath-self-mismatch.rs:8:5
   |
LL |     <i32 as Add<i32>>::add(1, 2u32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^-^^^^^^^
   |                            |
   |                            expected `i32`, found `i32`
   |
help: provide an argument of the correct type
   |
LL |     (1, {i32});
   |     ^^^^^^^^^^

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
